compiler/rustc_codegen_llvm/src/asm.rs

```diff
@@ -1029,6 +1029,15 @@ fn llvm_fixup_input<'ll, 'tcx>(
                 _ => value,
             }
         }
+        (InlineAsmRegClass::RiscV(RiscVInlineAsmRegClass::freg), Abi::Scalar(s))
+            if s.primitive() == Primitive::Float(Float::F16) =>
+        {
+            // Smaller floats are always "NaN-boxed" inside larger floats on RISC-V.
+            let value = bx.bitcast(value, bx.type_i16());
+            let value = bx.zext(value, bx.type_i32());
+            let value = bx.or(value, bx.const_u32(0xFFFF_0000));
+            bx.bitcast(value, bx.type_f32())
```

Comment:
Does LLVM not support using `f16` directly here? I expect this would result in much better codegen.

Reply:
My reading of the LLVM code is that it will accept `f16` directly if the zfhmin feature is enabled for the current function. Can you use `f16` directly in this case and only do the manual conversion to/from f32 if zfhmin is not available?